docs: improve compile from source docs

jkirschner-hashicorp opened the pull request:
Closes #10629, in which a practitioner had trouble figuring out how to build `consul` from source.

**Changes:**
- docs: link from makefile to compile instructions
  - Some practitioners look to the makefile directly rather than to the consul website for information on how to compile from source. Link to the website instructions directly from the makefile so the practitioner can accomplish their task successfully without a careful read of the makefile.

- docs: show how to cross-compile from source

**Installation Page Screenshot**

The instructions for compiling on the current system are the default tab. This is what the cross-compile tab looks like. The red line shows an indentation issue that can't be fixed without changing the pre-commit linter rules (see comment added in PR diff).

![image](https://user-images.githubusercontent.com/85913323/129103842-d0c0de8c-2067-4e64-a3b2-e239ad9f86e4.png)

jkirschner-hashicorp commented on website/content/docs/install/index.mdx:
````diff
@@ -56,14 +56,44 @@ a copy of [`git`](https://www.git-scm.com/) in your `PATH`.
-   ```shell
-   $ make dev
-   ```
+<!--
````
Not sure there's a better way around the problem described here

blake replied:
I've run into this same problem with the pre-commit hook. `git commit --no-verify` will allow you to commit the content unmodified.

jkirschner-hashicorp replied:
Thanks Blake. Didn't know that was an option!

dnephin reviewed (commented):
Thanks for writing this up!

It makes me a little sad that we are hiding the standard `go` toolchain here.  I personally never use the makefile. If I need to build Consul I run `go build`.  If I want to cross compile I can run `GOOS=darwin GOARCH=arm go build`.

The last time I looked, the makefile goes through a bunch of bash, but it doesn't actually change anything about the `go build` command. We don't really need any special flags, and we don't have any pre-compile steps to run.

I think there are advantages to being able to use the Go toolchain directly. It makes the entire process of building and testing Consul feel less magical, and it signals to contributors that they can expect to use the Go toolchain the way they would on any other project.

I would love it if we documented just using `go build`, but if others feel there is still advantages to the makefile, I don't want to block merging this. We can discuss further in another forum.

jkirschner-hashicorp commented:
@dnephin : looks like we now need a different reviewer to look at your build system changes :) so I added the Core team as a reviewer.

jkirschner-hashicorp commented on website/content/docs/install/index.mdx:
```diff
@@ -38,32 +38,53 @@ command-line, make sure to place it somewhere on your `PATH`.
 To compile from source, you will need [Go](https://golang.org) installed and
-configured properly (including a `GOPATH` environment variable set), as well as
-a copy of [`git`](https://www.git-scm.com/) in your `PATH`.
+configured properly, as well as a copy of [`git`](https://www.git-scm.com/) in your `PATH`.
```
@dnephin : what does "configured properly" mean (beyond installing it)?

dnephin replied:
Good question, that text was there before I edited.

I suspect it means "$PATH should include the path to the git binary", or "the git binary must be in $PATH".

kisunji replied:
I think it's clearer to remove "and configured properly" since it ambiguously implies a "proper" configuration (we're fine with any kind of defaults from go installation).

kisunji commented on website/content/docs/install/index.mdx:
````diff
@@ -38,32 +38,53 @@ command-line, make sure to place it somewhere on your `PATH`.
+```
+
+</Tab>
+<Tab heading="Cross-compile for another systems">
````
"another system" or "other systems"?

kisunji commented on GNUmakefile:
```diff
@@ -133,20 +134,17 @@ ifdef SKIP_DOCKER_BUILD
 # used to make integration dependencies conditional
 noop: ;
 
-bin: tools
```
@dnephin is `tools` no longer required?

dnephin replied:
I believe that's correct. We shouldn't require any additional tools to build the binary. `tools` installs stuff for updating protos , re-generating the bindata (which you are removing), run lint, or generate mocks, but nothing that should be required to build the binary.

jkirschner-hashicorp commented:
@kisunji : addressed your two minor points of feedback. Ready for re-review.

kisunji commented:
> @kisunji : addressed your two minor points of feedback. Ready for re-review.

Could you try rebasing? I updated the CircleCI images and found that they had more restrictive permissions so I _think_ some stages will fail in this PR

jkirschner-hashicorp commented:
@kisunji : just rebased (and addressed the tabbing issue based on @blake 's suggestion). Can you:
- Check whether the CI build works as you'd expect
- Authorize the Vercel deplyoment? (I can't do it for some reason)

kisunji commented on .circleci/config.yml:
```diff
@@ -433,7 +439,11 @@ jobs:
+          name: Build
+          command: |
+            make dev
+            mkdir -p /go/bin
+            cp ./bin/consul /go/bin/consul
```
```suggestion
            mkdir -p /home/circleci/go/bin
            cp ./bin/consul /home/circleci/go/bin/consul
```
```suggestion
            mkdir -p /go/bin
            cp ./bin/consul /go/bin/consul
```

jkirschner-hashicorp replied:
Change made and branch updated.

hc-github-team-consul-core commented:
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/574457.

hc-github-team-consul-core commented:
:cherries::white_check_mark: Cherry pick of commit 1ff27b0b16956431d591309487d5003c1ca02bb3 onto `stable-website` succeeded!

hc-github-team-consul-core commented:
:cherries::white_check_mark: Cherry pick of commit 1ff27b0b16956431d591309487d5003c1ca02bb3 onto `release/1.11.x` succeeded!
